azure-pipelines.yml

```diff
@@ -3,19 +3,46 @@ trigger:
 - release/*
 
 jobs:
-  - job: Linux
+  - job: CMake
     timeoutInMinutes: 0
-    pool:
-      vmImage: 'ubuntu-latest'
     strategy:
       matrix:
-        debug:
+        Linux.Debug:
+          image_name: 'ubuntu-latest'
+          deps: 'sudo apt-get install -y ninja-build clang libicu-dev'
+          build_type: 'Debug'
+          test_target: 'check'
+        Linux.ReleaseWithDebug:
+          image_name: 'ubuntu-latest'
+          deps: 'sudo apt-get install -y ninja-build clang libicu-dev'
+          build_type: 'RelWithDebInfo'
+          test_target: 'check'
+        Linux.Release:
+          image_name: 'ubuntu-latest'
+          deps: 'sudo apt-get install -y ninja-build clang libicu-dev'
+          build_type: 'Release'
+          test_target: 'all'
+        OSX.Debug:
+          image_name: 'macOS-latest'
+          deps: 'brew install ninja icu4c'
```

Comment:
Can we add a flag to each config for static vs shared?

(And obviously need to set OSX.Debug to static so its tests will pass)

Reply:
Yep, that is in the new commit.